File: src/main/java/com/vaadin/demo/component/tabs/TabsHideScrollButtons.java
package com.vaadin.demo.component.tabs;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.tabs.Tab;
import com.vaadin.flow.component.tabs.Tabs;
import com.vaadin.flow.component.tabs.TabsVariant;
import com.vaadin.flow.router.Route;

@Route("tabs-hide-scroll-buttons")
public class TabsHideScrollButtons extends Div {

	public TabsHideScrollButtons() {
		// tag::snippet[]
		Tab analytics = new Tab("Analytics");
		Tab customers = new Tab("Customers");
		Tab dashboards = new Tab("Dashboards");
		Tab documents = new Tab("Documents");
		Tab orders = new Tab("Orders");
		Tab products = new Tab("Products");
		Tab tasks = new Tab("Tasks");

		Tabs tabs = new Tabs(
			analytics, customers, dashboards, documents, orders, products, tasks
		);
		tabs.addThemeVariants(TabsVariant.LUMO_HIDE_SCROLL_BUTTONS);
		// end::snippet[]
		add(tabs);
	}

	public static class Exporter extends DemoExporter<TabsHideScrollButtons> { // hidden-source-line
	} // hidden-source-line
}

File: frontend/demo/component/tabs/tabs-hide-scroll-buttons.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/tabs';

@customElement('tabs-hide-scroll-buttons')
export class Example extends LitElement {
  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-tabs theme="hide-scroll-buttons" style="max-width: 100%; width: 400px;">
        <vaadin-tab>Analytics</vaadin-tab>
        <vaadin-tab>Customers</vaadin-tab>
        <vaadin-tab>Dashboards</vaadin-tab>
        <vaadin-tab>Documents</vaadin-tab>
        <vaadin-tab>Orders</vaadin-tab>
        <vaadin-tab>Products</vaadin-tab>
        <vaadin-tab>Tasks</vaadin-tab>
      </vaadin-tabs>
      <!-- end::snippet[] -->
    `;
  }
}
